xgi/readwrite/xgi_data.py

```diff
@@ -113,37 +120,23 @@ def _request_from_xgi_data(dataset=None):
     Raises
     ------
     XGIError
-        If HTTP request is not successful of the dataset does not exist.
+        If the HTTP request is not successful or the dataset does not exist.
 
     See also
     ---------
     load_xgi_data
     """
 
     index_url = "https://gitlab.com/complexgroupinteractions/xgi-data/-/raw/main/index.json?inline=false"
-    r = requests.get(index_url)
-
-    if r.ok:
-        index_data = r.json()
-
-        if dataset is None:
-            print("Available datasets are the following:")
-            print(*index_data, sep="\n")
-            return
-    else:
-        raise XGIError(f"Error: HTTP response {r.status_code}")
+    index_data = _request_json_from_url(index_url)
```

Comment:
this line does not print the available datasets if dataset is None anymore, right?
is it printed in lines 134-137 instead?

Reply:
Correct. The list of available datasets is printed in lines 57-62 and I added a comment to reflect that. It got too messy when I tried to keep in in the `_request_json_from_url` function.